casper/src/main/scala/coop/rchain/casper/dag/BlockDagKeyValueStorage.scala

```diff
@@ -233,6 +241,32 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
   override def addDeploy(d: Signed[DeployData]): F[Unit] = deployStore.put(d.sig, d)
 
   override def pooledDeploys: F[Map[DeployId, Signed[DeployData]]] = deployStore.toMap
+
+  // Map of deploys being executed and execution results
+  private val execMap    = TrieMap.empty[DeployId, Option[String]]
+  private val expiredMap = TrieMap.empty[DeployId, Unit]
+  def deployStatus(d: DeployId): F[String] =
```

Comment:
Why is `deployStatus` not part of the `BlockDagStorage` trait?

Reply:
As @tgrospic specified, we don't want this to be on a trait but just a part of one implementation (BlockDagKeyValueStore).

Reply:
`deployStatus` which returns string is just temporary functions and should not be on block dag storage at all.

We need to see what is needed for BlockApi to return deploy status and add just that to block dag storage.